RFC-011-SWAP-Omni-Layer-basic.md

```diff
@@ -0,0 +1,320 @@
+# Omni Layer Basic HTLC Atomic Swap
+
+- RFC-Number: 011
+- Status: Draft
+- Discussion-issue: -
+- Created on: 27 Mar. 2019
+
+**Table of Contents**
+
+- [Description](#description)
+- [Limitations](#limitations)
+- [Ownership](#ownership)
+- [Inputs & Outputs format](#inputs--outputs-format)
+  * [Omni Layer data](#omni-layer-data)
+  * [Dust](#dust)
+  * [Deployment transaction](#deployment-transaction)
+  * [Redeem transaction](#redeem-transaction)
+  * [Refund transaction](#refund-transaction)
+- [Examples/Test vectors](#examplestest-vectors)
+  * [RFC003 SWAP REQUEST](#rfc003-swap-request)
+  * [RFC003 SWAP RESPONSE](#rfc003-swap-response)
+  * [HTLC](#htlc)
+
+## Description
+
+This RFC defines how to execute a [RFC003](./RFC-003-SWAP-basic.md) SWAP where one of the ledgers is Bitcoin and the associated asset is an Omni Layer token asset.
+
+For the definition of the Bitcoin ledger see [RFC004](./RFC-004-SWAP-Bitcoin.md).
+For the definition of the Omni Layer token asset see [RFC010](./TODO).
+
+This RFC re-purposes the Hash Time Locked Contract (HTLC) Bitcoin script from [RFC005](./RFC-005-SWAP-Bitcoin-basic.md) to lock Omni Layer tokens instead of Bitcoin.
+
+To fulfil the requirements of [RFC003](./RFC-003-SWAP-basic.md) this RFC defines:
+
+- How to construct Omni layer transactions transferring Omni Layer tokens to a HTLC address
+- how to deploy, redeem and refund the HTLC during the execution phase of [RFC003](./RFC-003-SWAP-basic.md)
+
+## Constructing an Omni Layer Transaction
+
+This sections describes how to build Omni Layer transactions for the purpose of a basic swap.
+Information from the [Omni Layer Spec](https://github.com/OmniLayer/spec) has been transcribed here to facilitate the description of a Omni Layer Basic HTLC Atomic Swap.
+
+This RFC only supports the [simple send](https://github.com/OmniLayer/spec#transfer-coins-simple-send) Omni Layer transaction type, encoded using Class C format.
+Class C transactions uses Bitcoin `OP_RETURN` instruction to store the Omni Layer data.
+
+For clarity purpose, an *input, output owning an Omni Layer token* refers to the fact that such input or output matches an address that owns such token.
+See [RFC-010](TODO#ownership) for more details.
```

Comment:
I'm just highlighting this TODO here so that we don't forget it.

Reply:
I can fix these todos know that #50 is merged.